### PR TITLE
operator: Add a PodDisruptionBudget to lokistack-gateway

### PR DESCRIPTION
**Pull request by aminesnow**
**What this PR does / why we need it**:
This PR adds a `PodDisruptionBudget` to the Loki Operator Gateway, with a `MinAvailable` of 1.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: operator/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ## Main
+- [9162](https://github.com/grafana/loki/pull/9162) **aminesnow**: Add a PodDisruptionBudget to Loki Operator Gateway
```
Comment (periklis) [inline operator/CHANGELOG.md:2]:
```suggestion
- [9162](https://github.com/grafana/loki/pull/9162) **aminesnow**: Add a PodDisruptionBudget to lokistack-gateway
```

**Review by periklis — APPROVED** [review]
Just nits

**Review by periklis — CHANGES_REQUESTED** [review]
Doesn't that need to run `make bundle-all`?

**Comment by aminesnow** [comment]
> Doesn't that need to run `make bundle-all`?

Yes it does.

**Review by periklis — APPROVED** [review]

**Pull request merged**